modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/util/IntrospectionUtils.java

```diff
@@ -571,20 +572,33 @@ public static String getMemberName(EnrichableModel enrichableModel, String defau
     return emptyList();
   }
 
-  public static List<ResolvableType> getInterfaceGenerics(final java.lang.reflect.Type type,
-                                                          final Class<?> implementedInterface) {
+  public static List<java.lang.reflect.Type> getInterfaceGenerics(final java.lang.reflect.Type type,
+                                                                  final Class<?> implementedInterface) {
 
     ResolvableType interfaceType = null;
     ResolvableType searchClass = ResolvableType.forType(type);
 
-    if (searchClass.getRawClass().equals(implementedInterface)) {
-      return asList(ResolvableType.forType(type).getGenerics());
+    if (implementedInterface.equals(searchClass.getRawClass())) {
+      return stream(searchClass.getGenerics()).map(resolvableType -> resolvableType.getType()).collect(toList());
     }
 
+    Map<String, java.lang.reflect.Type> genericTypes = new HashMap<>();
+
+    addGenericsToMap(genericTypes, searchClass);
+
     while (!Object.class.equals(searchClass.getRawClass())) {
       for (ResolvableType iType : searchClass.getInterfaces()) {
         if (implementedInterface.isAssignableFrom(iType.getRawClass())) {
+          addGenericsToMap(genericTypes, iType);
           interfaceType = iType;
+          while (!implementedInterface.equals(interfaceType.getRawClass())) {
+            for (ResolvableType innerIType : interfaceType.getInterfaces()) {
+              if (implementedInterface.isAssignableFrom(innerIType.getRawClass())) {
```

Comment:
`while -> for ->  if ->  while  ->  for ->  if` doesn't sound like a good idea. Any chance we refactor this?